libs/sdk-core/src/breez_services.rs

```diff
@@ -473,14 +493,56 @@ impl BreezServices {
         pair_hash: String,
     ) -> Result<ReverseSwapInfo> {
         let routing_hop = self.lsp_info().await?;
-        self.btc_send_swapper
+        let created_reverse_swap_info = self
+            .btc_send_swapper
             .create_reverse_swap(
                 amount_sat,
                 onchain_recipient_address.clone(),
                 pair_hash,
                 routing_hop.pubkey,
             )
-            .await
+            .await?;
+        info!("Created reverse swap: {created_reverse_swap_info:?}");
+
+        // Wait until one of the following happens:
+        // - trying to pay the HODL invoice explicitly fails from Greenlight
+        // - the regular poll of the Breez API detects the status of this reverse swap advanced to LockTxMempool
+        //   (meaning Boltz detected that we paid the HODL invoice)
+        // - the max allowed duration of a payment is reached
+        tokio::select! {
+            pay_thread_res = tokio::time::timeout(
+                Duration::from_secs(self.node_api.config().payment_timeout_sec as u64),
+                self.node_api.send_payment(created_reverse_swap_info.hodl_bolt11.clone(), None)
+            ) => {
+                // TODO It doesn't fail when trying to pay more sats than max_payable?
+                match pay_thread_res {
+                    // Paying a HODL invoice does not typically return, so if send_payment() returned, it's an abnormal situation
+                    Ok(Ok(res)) => Err(anyhow!("Payment of HODL invoice unexpectedly returned: {res:?}")),
+
+                    // send_payment() returned an error, so we know paying the HODL invoice failed
+                    Ok(Err(e)) => Err(anyhow!("Failed to pay HODL invoice: {e}")),
+
+                    // send_payment() has been trying to pay for longer than the payment timeout
+                    Err(e) => Err(anyhow!("Trying to pay the HODL invoice timed out: {e}"))
+                }
+            },
+            paid_invoice_res = self.poll_initial_boltz_status_transition(&created_reverse_swap_info.id) => {
+                match paid_invoice_res {
+                    Ok(_) => Ok(created_reverse_swap_info),
+                    Err(e) => Err(anyhow!(e))
+                }
+            }
+        }
+    }
+
+    /// Returns an optional in-progress [ReverseSwapInfo]
+    pub async fn in_progress_reverse_swap(&self) -> Result<Option<ReverseSwapInfo>> {
+        let monitored = self.btc_send_swapper.list_monitored()?;
```

Comment:
Not sure all monitored are also defined as "in progress".
"in progress" is a swap that blocks other swaps from starting and I remember we were discussing the fact that the last confirmation step should be still monitored but will not block a new swap.

Reply:
Done 50e9b212dc471c50a93480de16b24f2acbdefd3a